tests/tests_fabric/strategies/test_fsdp_integration.py

```diff
@@ -161,6 +166,34 @@ def test_fsdp_save_full_state_dict(tmp_path):
     params_after = list(fabric.model.parameters())
     assert all(torch.equal(p0, p1) for p0, p1 in zip(params_before, params_after))
 
+    # run a step to verify the optimizer state is correct
+    fabric.run()
+
+    # verify the full state can be loaded back into a FSDP model/strategy
+    fabric = BoringFabric(
+        accelerator="cuda",
+        strategy=FSDPStrategy(auto_wrap_policy=always_wrap_policy, state_dict_type="full"),
+        devices=2,
+    )
+    fabric.run()
+    metadata = fabric.load(checkpoint_path, {"model": fabric.model, "optimizer": fabric.optimizer})
+    assert metadata == {"steps": 1}
```

Comment:
I think this block is redundant since we are doing the same above. Could we combine the asserts for the optimizer with the code lines above? Btw, for loading, passing `state_dict_type="full"` is not necessary.

Reply:
I agree with your suggestion that using `state_dict_type="full"` is unnecessary. 

The purpose of the first mentioned block is to ensure that a model can load states saved from FSDP without FSDP (such as DDP). On the other hand, the second-mentioned block is responsible for restoring FSDP's weights. 

In the `update better unit test` commit, I added another test, so now we have:
- Save from FSDP, restore to FSDP
- Save from FSDP, restore to other strategy
- Save from other strategy, restore to FSDP